### ISSUE TITLE
rake db:schema:dump fails with schema_plus

### ISSUE THREAD

**Issue by thomassnielsen** [SchemaPlus/schema_plus_core#4]
We've got a pretty complex database that will be the base of our new rails-based API. Running `rake db:schema:dump` works as expected normally, but after adding the `schema_plus_views` gem, it fails with the stack trace below. 

We've tried digging a little, but haven't found the source of the issue. Any ideas what might cause the issue?

```
❯ rake db:schema:dump --trace
** Invoke db:schema:dump (first_time)
** Invoke environment (first_time)
** Execute environment
** Invoke db:load_config (first_time)
** Execute db:load_config
** Execute db:schema:dump
rake aborted!
NoMethodError: undefined method `[]' for nil:NilClass
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/schema_plus_core-0.5.1/lib/schema_plus/core/active_record/schema_dumper.rb:63:in `block in table'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/modware-0.1.2/lib/modware/stack.rb:55:in `call'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/modware-0.1.2/lib/modware/stack.rb:55:in `call_implementation'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/modware-0.1.2/lib/modware/stack.rb:85:in `_modware_continue'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/modware-0.1.2/lib/modware/stack.rb:77:in `_modware_call'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/modware-0.1.2/lib/modware/stack.rb:44:in `execute'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/modware-0.1.2/lib/modware/stack.rb:20:in `start'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/schema_monkey-2.1.1/lib/schema_monkey/stack.rb:34:in `start'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/schema_plus_core-0.5.1/lib/schema_plus/core/active_record/schema_dumper.rb:48:in `table'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/activerecord-4.2.3/lib/active_record/schema_dumper.rb:97:in `block in tables'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/activerecord-4.2.3/lib/active_record/schema_dumper.rb:96:in `each'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/activerecord-4.2.3/lib/active_record/schema_dumper.rb:96:in `tables'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/schema_plus_core-0.5.1/lib/schema_plus/core/active_record/schema_dumper.rb:43:in `block in tables'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/modware-0.1.2/lib/modware/stack.rb:55:in `call'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/modware-0.1.2/lib/modware/stack.rb:55:in `call_implementation'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/modware-0.1.2/lib/modware/stack.rb:85:in `_modware_continue'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/modware-0.1.2/lib/modware/stack.rb:77:in `_modware_call'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/modware-0.1.2/lib/modware/stack.rb:83:in `_modware_continue'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/modware-0.1.2/lib/modware/stack.rb:77:in `_modware_call'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/modware-0.1.2/lib/modware/stack.rb:44:in `execute'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/modware-0.1.2/lib/modware/stack.rb:20:in `start'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/schema_monkey-2.1.1/lib/schema_monkey/stack.rb:34:in `start'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/schema_plus_core-0.5.1/lib/schema_plus/core/active_record/schema_dumper.rb:42:in `tables'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/activerecord-4.2.3/lib/active_record/schema_dumper.rb:38:in `dump'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/schema_plus_core-0.5.1/lib/schema_plus/core/active_record/schema_dumper.rb:17:in `dump'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/activerecord-4.2.3/lib/active_record/schema_dumper.rb:22:in `dump'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/activerecord-4.2.3/lib/active_record/railties/databases.rake:237:in `block (4 levels) in <top (required)>'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/activerecord-4.2.3/lib/active_record/railties/databases.rake:236:in `open'
/Users/thomas/.rvm/gems/ruby-2.2.1/gems/activerecord-4.2.3/lib/active_record/railties/databases.rake:236:in `block (3 levels) in <top (required)>'
/Users/thomas/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/rake/task.rb:240:in `call'
/Users/thomas/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/rake/task.rb:240:in `block in execute'
/Users/thomas/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/rake/task.rb:235:in `each'
/Users/thomas/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/rake/task.rb:235:in `execute'
/Users/thomas/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/rake/task.rb:179:in `block in invoke_with_call_chain'
/Users/thomas/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/monitor.rb:211:in `mon_synchronize'
/Users/thomas/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/rake/task.rb:172:in `invoke_with_call_chain'
/Users/thomas/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/rake/task.rb:165:in `invoke'
/Users/thomas/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/rake/application.rb:150:in `invoke_task'
/Users/thomas/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/rake/application.rb:106:in `block (2 levels) in top_level'
/Users/thomas/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/rake/application.rb:106:in `each'
/Users/thomas/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/rake/application.rb:106:in `block in top_level'
/Users/thomas/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/rake/application.rb:115:in `run_with_threads'
/Users/thomas/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/rake/application.rb:100:in `top_level'
/Users/thomas/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/rake/application.rb:78:in `block in run'
/Users/thomas/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/rake/application.rb:176:in `standard_exception_handling'
/Users/thomas/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/rake/application.rb:75:in `run'
/Users/thomas/.rvm/rubies/ruby-2.2.1/bin/rake:33:in `<main>'
/Users/thomas/.rvm/gems/ruby-2.2.1/bin/ruby_executable_hooks:15:in `eval'
/Users/thomas/.rvm/gems/ruby-2.2.1/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by ronen** [comment]
@thomassnielsen Coincidentally, just a few days ago @stenver raised a issue over in SchemaPlus/schema_plus#224, which had a crash at the same line of code.

In that case, it was due to a table being defined using a custom data type for one of the columns.  ActiveRecord couldn't dump the table so emitted an error message as a comment to the dump stream, and schema_plus_core choked trying to parse that message as a table definition.  

I've just released schema_plus_core v0.6.0 with that fixed; any time schema_plus_core can't parse whatever ActiveRecord emitted for a table definition, schema_plus_core just passes it along.  I expect that this should prevent your crash; though I also expect that (with or without schema_plus_core) your dump will include error message comments for the tables it couldn't dump.

Let me know if v0.6.0 does indeed help...
